### PR TITLE
[8.x] Better ScriptHandler.php, migrate from drustack/drustack-standard

### PR DESCRIPTION
**Pull request by hswong3i**
Migrate from https://github.com/drustack/drustack-standard/blob/master/src/Composer/ScriptHandler.php, include:
- Original script handlers, clone from Drupal 8.x Drupal\Core\Composer\Composer
- Generate info meta data into .info, logic clone from drush 8.x

### CODE REVIEW

**Comment by webflo** [comment]
@hswong3i Wrong upstream repo?

**Comment by hswong3i** [comment]
Sorry I remove the create web/modules web/themes and web/profiles code from script handler, so the test get failed (because I already remove that from my https://github.com/drustack/drustack-standard/blob/master/src/Composer/ScriptHandler.php)

Latest commit try to create those folder manually ;-)

**Comment by hswong3i** [comment]
Progress also backport and contribute to Drupal core, please kindly give a hand if possible:
- https://www.drupal.org/node/2755739
- https://www.drupal.org/node/2755741
- https://www.drupal.org/node/2755745

File: composer.json [hunk 1]
```diff
@@ -1,14 +1,42 @@
...
-    "description": "Project template for Drupal 8 projects with composer",
-    "type": "project",
-    "license": "GPL-2.0+",
     "authors": [
         {
             "name": "",
             "role": ""
         }
     ],
+    "autoload": {
```
Comment (hussainweb) [inline composer.json:8]: I like the original ordering better. This is just alphabetical but it is of little use. I think this better remain with contextual ordering.

Comment (hswong3i) [inline composer.json:8]: The reformat is mainly due to running `:%!python -m json.tool` inside vim ;-)

**Comment by hswong3i** [comment]
Some update due to https://www.drupal.org/node/2760467 changes.

**Comment by webflo** [comment]
Closing this because its a huge change i won't copy the vendor cleanup script and info rewrite. We could create seperate plugins for it.

**Comment by webflo** [comment]
Thanks for your work. Please review https://github.com/drupal-composer/info-rewrite or https://www.drupal.org/project/composer_deploy

**Pull request closed without merging**